### PR TITLE
fix: XYNE-63092 include mentions in canvas search

### PR DESCRIPTION
**Pull request by uayush01**
## Type of Change

- [x] Bugfix

## Description

Fixes XYNE-63092.

Canvas Cmd+F now opens Canvas search while viewing a canvas, instead of falling through to the global command/search palette. Canvas search also includes inline tagged mentions by reading mention display props, so tagged users/groups are counted like visible text.

## Areas Touched

- [x] `apps/dashboard`

---

## Zero (Sync) Changes

- [x] No Zero changes — **skip this section**

## Schema & Data Changes

- [x] No schema changes — **skip this section**

## Config, Environment & URLs

- [x] No config changes — **skip this section**

## API Contract

- [x] No API changes

---

### Automated

- [x] Not covered by tests
  - Small UI shortcut/search utility fix; verified with lint/typecheck and local manual smoke.

Ran:
- `pnpm --filter xyne-spaces-dashboard exec eslint --quiet src/components/Canvas/CanvasScreen/CanvasScreen.tsx src/utils/searchUtils.ts`
- `pnpm --filter xyne-spaces-dashboard run typecheck`

### Manual

**Run mode**
- [x] Local dev (`pnpm run dev:all`)

**Surface & data**
- [x] Web browser

Manual smoke:
- Opened local Canvas with plain text and tagged mention.
- Verified Cmd+F opens Canvas search instead of global palette.
- Verified searching a tagged user name is included in Canvas search matches.

---

## Checklist

- [x] Reviewed my own diff; scoped to one thing
- [x] Commits follow `<type>: <TICKET-ID> <subject>`; branch is `fix/*` or `feature/*`
- [x] No debug logging or commented-out code left behind

POT : https://drive.google.com/file/d/1bEDXkvl-Nf0eMIJiowU99CZ0RCnZAuEJ/view?usp=sharing

### CODE REVIEW

File: apps/dashboard/src/utils/searchUtils.ts [hunk 1]
```diff
@@ -56,6 +60,15 @@ const getBlockText = (content: BlockContent[]): string => {
     .join('');
 };
 
+const getMentionDisplayText = (item: BlockContent): string => {
+  const props = item.props || {};
+  const groupId = typeof props['groupId'] === 'string' ? props['groupId'] : '';
+  const groupName = typeof props['groupName'] === 'string' ? props['groupName'] : '';
+  const username = typeof props['username'] === 'string' ? props['username'] : '';
+
+  return groupId && groupName ? groupName : username;
```
Comment (XyneSpaces) [inline apps/dashboard/src/utils/searchUtils.ts]: **[major] R5 — duplicate logic / drift risk** — `getMentionDisplayText` re-implements the renderer's display-name logic from `CanvasMentionSpec.tsx` (`MentionRender`: `props.groupId && props.groupName ? props.groupName : props.username || ''`). Today the two copies match exactly, but they will drift: the next change to the mention spec (e.g. adding a `userEmail` or display-name fallback) silently desyncs the text that search matches from the text actually rendered — precisely the search↔render alignment this PR fixes. ↳ Suggested fix: export one shared helper (e.g. in a mention-utils module) consumed by both `MentionRender` and `getBlockText`, so search and render cannot diverge.

File: apps/dashboard/src/utils/searchUtils.ts [hunk 2]
```diff
@@ -42,6 +43,9 @@ const getBlockText = (content: BlockContent[]): string => {
 
   return content
     .map(item => {
+      if (item.type === 'mention') {
+        return getMentionDisplayText(item);
```
Comment (XyneSpaces) [inline apps/dashboard/src/utils/searchUtils.ts]: **[major] R7 — new behavior with no tests** — this PR adds three distinct testable behaviors to a pure utility (mention text extraction incl. the group-vs-user fallback in `getMentionDisplayText`; end-of-text positioning in `getDomPosition`; avatar-initials exclusion via `isSearchableTextNode`), and the PR self-declares "Not covered by tests". `searchUtils.ts` is trivially unit-testable and has no existing test file. ↳ Suggested fix: add unit tests before merge covering group mention, user mention, missing/empty props, `targetOffset === total` (end-of-text), and a text node inside `[data-slot="avatar"]` (guideline 204 — unit tests for new functionality/edge cases).

File: apps/dashboard/src/components/Canvas/CanvasScreen/CanvasScreen.tsx [hunk 3]
```diff
@@ -432,6 +435,7 @@ const CanvasScreen: React.FC<CanvasScreenProps> = ({
     }
   }, [
     singleCanvas,
+    singleCanvasDetails.type,
```
Comment (XyneSpaces) [inline apps/dashboard/src/components/Canvas/CanvasScreen/CanvasScreen.tsx:438]: **[minor] R2 — partial dependency tracking (assumption)** — the dep array adds `singleCanvasDetails.type` while `singleCanvasDetails` itself is not a dependency. If the effect body reads any other property of that object (the body is outside this diff's context — assumption), those reads are untracked and can serve stale values. ↳ Suggested fix: list the specific fields the effect actually reads (or the whole object if it is referentially stable) instead of cherry-picking a single field.

**Review by XyneSpaces — COMMENTED** [review]
Automated review per the Xyne Spaces Review Guidelines — 2 files, 10 hunks checked against the full rule set. Findings: 2 major, 1 minor (details inline). Full summary + reviewer recommendation follows in a top-level comment.

**Comment by XyneSpaces** [comment]
## Xyne Spaces Reviewer — PR #1695

**Verdict:** Approve-worthy after addressing the two major findings (shared mention-display helper + tests)
**Reviewed:** 2 files, 10 hunks · **Findings:** 0 blocking, 2 major, 1 minor

### Findings
| Severity | Rule | Location | Issue |
|----------|------|----------|-------|
| major | R5 | `apps/dashboard/src/utils/searchUtils.ts:69` | `getMentionDisplayText` re-implements `MentionRender`'s display-name logic (`CanvasMentionSpec.tsx`) — the two copies match today, but any future change to the mention spec silently desyncs search text from rendered text |
| major | R7 | `apps/dashboard/src/utils/searchUtils.ts:47` | Three new testable behaviors (mention extraction with group/user fallback, end-of-text DOM positioning, avatar-initials exclusion) ship with zero tests; the PR self-declares "Not covered by tests" |
| minor | R2 | `apps/dashboard/src/components/Canvas/CanvasScreen/CanvasScreen.tsx:438` | Dep array tracks `singleCanvasDetails.type` but not the object itself — other fields read inside the effect would go untracked *(assumption — effect body is outside the diff context)* |

**Verified while reviewing:** the mention prop shape (`groupId`/`groupName`/`username`) and fallback logic exactly match `CanvasMentionSpec.tsx`'s `MentionRender`; the `[data-slot="avatar"]` exclusion matches `Avatar.tsx` (initials render inside AvatarRoot; group/no-access icons are textless SVGs); `getDomPosition`'s end-of-text return fires only when `targetOffset === total` and the in-loop return still covers all earlier offsets. No migration/schema files touched → migration-safety gate N/A.

### Recommended reviewer
**@VishalM-23** (@mentioned here — the formal reviewer-request API isn't available to this automation): closest prior work on search + the mention data model, and active in Spaces today.

| Candidate | Domain evidence in this PR's area | Default reviewer? | Pending reviews | Tie-breaker |
|-----------|-----------------------------------|-------------------|-----------------|-------------|
| @VishalM-23 | #1458 search side panel; #933 DM search + mention pickers | no (no CODEOWNERS cover these paths) | not computable* | active in Spaces today |
| @utkarsh-611 | deepest canvas-editor background | no | not computable* | last seen in Spaces 12 May 2026 |
| @MukherjeeSourish | #856/#858 search palette | no | not computable* | last seen 26 Aug 2026 |
| nasimshk | #1258 cmd+K search filter registry | no | not computable* | search-infra adjacent; this PR's core is the mention prop model, where #933 is the stronger fit |

\* Signals that could not be computed honestly: commits-by-path (this toolset has no path-filtered commit listing) and per-reviewer pending workload (`review-requested:<login>` search returned ~all 296 open PRs regardless of login — discarded as unreliable). The repo has no CODEOWNERS entries covering `apps/dashboard/src/utils/searchUtils.ts` or the CanvasScreen path, and the PR has no requested reviewers, so ranking fell back to PR-history domain fit → Spaces activity. PR author @uayush01 and the XyneSpaces bot account are excluded.

@VishalM-23 — your #933 (DM search + mention pickers) and #1458 (search side panel) are the closest prior work to this mention-in-search fix; could you take this review?

**Review by harshpreet931 — APPROVED** [review]

**Review by Mohan0544 — APPROVED** [review]
